catalog/spatial/geocoding/spatial-geocoding-admin-module/Gruntfile.js

```diff
@@ -0,0 +1,173 @@
+                timeout: 1000 * 30,
+                slow: 10000
+            },
+            phantom: {
+                src: ['src/test/js/wd/*.js'],
+                options: {
+                    hostname: '127.0.0.1',
+                    usePhantom: true,
+                    phantomPort: 5555
+                }
+            },
+            selenium: {
+                src: ['src/test/js/wd/*.js'],
+                options: {
+                    // make sure to start selenium server at host:port first
+                    hostname: '127.0.0.1',
+                    port: 4444,
+                    // mochaWebdriver always starts a selenium server so
+                    // starting phantomjs instance that will not be used
+                    phantomPort: 5555,
+                    usePhantom: true
+                }
+            },
+            sauce: {
+                src: ['src/test/js/wd/*.js'],
+                options: {
+                    autoInstall: false,
+                    testName: 'Search UI',
+                    concurrency: 3,
+                    timeout: 1000 * 60 * 2,
+                    browsers: [
+                        {platform: 'Windows 7', browserName: 'internet explorer', version: '9'},
+                        {platform: 'Windows 7', browserName: 'chrome', version: '38'},
+                        {platform: 'Windows 7', browserName: 'firefox', version: '31'}
+                    ]
+                }
+            }
+        },
+        express: {
+            options: {
+                port: 8282,
+                hostname: '*'
+            },
+            test: {
+                options: {
+                    port: 8888,
+                    server: './test.js'
+                }
+            },
+            server: {
+                options: {
+                    server: './server.js'
+                }
+            }
+        },
+        watch: {
+            jsFiles: {
+                files: ['<%= jshint.all.src %>'],
+                tasks: ['newer:jshint']
+            },
+            livereload : {
+                options : {livereload :true},
+                files : ['target/webapp/css/index.css'
+                    // this one is more dangerous, tends to reload the page if one file changes
+                    // probably too annoying to be useful, uncomment if you want to try it out
+//                    '<%= jshint.files %>'
+                ]
+            },
+            cssFiles : {
+                files :['src/main/webapp/css/*.css'],
+                tasks : ['cssmin']
+            },
+            bowerFile: {
+                files: ['src/main/webapp/bower.json'],
+                tasks: ['bower']
+            }
+        }
+    });
+
+    grunt.registerTask('test', ['express:test', 'mochaWebdriver:phantom']);
+    grunt.registerTask('test:selenium', ['express:test', 'mochaWebdriver:selenium']);
+    grunt.registerTask('test:sauce', ['express:test', 'mochaWebdriver:sauce']);
```

Comment:
I don't see any UI tests for this module. We really should have some UI tests, even if they are just some basic unit tests for the models. However, it looks like we are still pulling in selenium and sauce, along with the express web server with the test.js, server.js, and server-imply.js files and then not doing anything with them.